src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -254,39 +255,54 @@ CHIP_ERROR InitCredentialSets()
     return CHIP_NO_ERROR;
 }
 
-} // namespace
+} // anonymous namespace
+
+class TestCASESession
```

Comment:
I'm actually not aware of other tests using this pattern. Why is it useful to have static methods instead of normal global functions?

Reply:
Could namespace it if we really want some isolation. CASE_ here is a bit smurf-named.

Reply:
Mainly derived this pattern from Interaction Model

src/app/tests/TestCommandInteraction.cpp
src/app/tests/TestInteractionModelEngine.cpp
src/app/tests/TestReadInteraction.cpp
src/app/tests/TestReportingEngine.cpp

There are other. While it isn't the most common paradigm, it is already used in the SDK albeit in `src/app` only at this point, but it is used when using `friend` to give the unit test access to some special test functions that are not public.

>Let's keep using functions instead of static

My understanding is we need static for `CASE_SimulateUpdateNOCInvalidatePendingEstablishment` since that needs to be encapsulated in a class that we can then use to allow the `friend` to allow access to non-public aspects of the class. For `CASESession` this allows us to keep the State as non public and makes sure that we are very explicit that test APIs for that class are only used by `TestCASESession` and no where else.

While we can keep using global functions for all other tests I think I personal thing the one file should be consistent and while I am doing this refactor to bring all the other tests inline

Reply:
>while I am doing this refactor to bring all the other tests inline

Just to clarify I meant that all other tests in this file to be inline with `SimulateUpdateNOCInvalidatePendingEstablishment`